### ISSUE TITLE
GitHub Workflow Running out of disk space

### ISSUE THREAD

**Issue by bap14** [wardenenv/images#24]
Looks like the PHP images are running out of disk space, and have been for a while now. Looks like we're limited to 14GB of disk space in a GitHub hosted worker, unless we host our own runner(s).

A few options available here: https://github.com/orgs/community/discussions/25678 but it seems like only one would apply to us (since we're using centos instead of debian images): remove /opt/hostedtoolcache

### Expected Result:
PHP images are built without issue

### Actual Result:
PHP images get mostly built but then run into an error with diskspace

**Comment by bap14** [comment]
@navarr Not sure of the root cause of this. Is it because we're building 5 base PHP images in one build step, or that we're not cleaning any caches in between? I think cleaning up the hosted tool cache is a stop-gap solution and this will eventually come back up again.

Another way to resolve this is to stop building once a PHP version reaches EOL. So for instance we would not build PHP 7.4 or 8.0 anymore, because there aren't updates to it. At the end of this year we'd stop building PHP 8.1 because it's having security releases stopped in November. While this may solve the problem, it seems like it's a bit of a larger discussion (what version(s) of PHP do we support, how long to we build them, etc.).

**Comment by navarr** [comment]
I don't think we can stop building at eol. Sadly developers still depend on those, as much as I'd love them to leave them. As long as it's not troublesome to maintain I'd prefer to maintain them.

That said, we could probably split these up a bit more. Den was very split up but it took a very long time to build.

**Comment by bap14** [comment]
@navarr Looks like base PHP images are building, so that's good. Looks like NPM is failing installing libraries or packages.

**Comment by bap14** [comment]
Looks like this is a potentially known issue with no resolution (other than potentially abandoning upgraded Node).

https://github.com/actions/runner-images/issues/3737

This brings back my suggestion of leaving unsupported versions behind. I get the images need to be available. The images should stick around even without pushing, but if PHP isn't receiving new updates, and Node 12, 14, 16, 18 aren't receiving updates, why are we rebuilding them? I'm not sure of the image retention period of each registry, but I can't imagine it's shorter than like 6 months. Just something to think about.
